src/hotspot/share/utilities/stableValue.hpp

```diff
@@ -0,0 +1,76 @@
+/*
+* Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+* DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+*
+* This code is free software; you can redistribute it and/or modify it
+* under the terms of the GNU General Public License version 2 only, as
+* published by the Free Software Foundation.
+*
+* This code is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+* version 2 for more details (a copy is included in the LICENSE file that
+* accompanied this code).
+*
+* You should have received a copy of the GNU General Public License version
+* 2 along with this work; if not, write to the Free Software Foundation,
+* Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+*
+* Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+* or visit www.oracle.com if you need additional information or have any
+* questions.
+*
+*/
+
+#ifndef SHARE_UTILITIES_STABLEVALUE_HPP
+#define SHARE_UTILITIES_STABLEVALUE_HPP
+
+#include "globalDefinitions.hpp"
+
+// A memory area with adequate size and alignment for storage of a T.
+// May be initialized exactly once.
+// The purpose of this class is to defer initialization of a T to a later point in time,
+// and then to never deallocate it. This is mainly useful for deferring the initialization of
+// static fields in classes, in order to avoid "Static Initialization Order Fiasco".
+template<typename T>
+class StableValue {
+  union {
+    T _t;
+  };
+
+  DEBUG_ONLY(bool _initialized);
+
+public:
+  NONCOPYABLE(StableValue);
+
+  StableValue()
+  DEBUG_ONLY(: _initialized(false)) {
+    // Do not construct value, on purpose.
+  }
+
+  ~StableValue() {
+    // Do not destruct value, on purpose.
+  }
+
+  T* get() {
+    assert(_initialized, "must be initialized before access");
+    return &_t;
+  }
+
+  T& operator*() {
+    return *get();
+  }
+
+  T* operator->() {
+    return get();
+  }
+
+  template<typename... Ts>
+  void initialize(Ts&... args) {
+    DEBUG_ONLY(_initialized = true);
```

Comment:
Shouldn't we also check here that `_initialized == false`?

Reply:
Yes, it should.  My (untested) proto earlier had the other, more serious, bug of forgetting to
assign `_initialized`!

Reply:
Yup